manager/controlapi/network.go

```diff
@@ -209,7 +209,10 @@ func (s *Server) removeNetwork(id string) error {
 		}
 
 		for _, t := range tasks {
-			if t.DesiredState <= api.TaskStateRunning && t.Status.State <= api.TaskStateRunning {
+			// we don't care about the desired state of the task, only its
```

Comment:
Actually this does not work because the service has likely already been deleted by now and so the function will return early.

Reply:
we probably should also put a unit-test for this fix.